src/transactions/test/ChangeTrustTests.cpp

```diff
@@ -268,6 +268,136 @@ TEST_CASE("change trust", "[tx][changetrust]")
 
     SECTION("pool trustline")
     {
+        auto getNumSubEntries = [&](AccountID const& accountID) {
+            LedgerTxn ltx(app->getLedgerTxnRoot());
+            auto acc = stellar::loadAccount(ltx, accountID);
+            return acc.current().data.account().numSubEntries;
+        };
+
+        auto poolShareTest = [&](Asset const& assetA, Asset const& assetB) {
+            // trustlines will be deauthorized initially
+            gateway.setOptions(setFlags(AUTH_REQUIRED_FLAG));
+
+            auto poolShareAsset = makeChangeTrustAssetPoolShare(
+                assetA, assetB, LIQUIDITY_POOL_FEE_V18);
+
+            // root is missing trustline(s)
+            REQUIRE_THROWS_AS(root.changeTrust(poolShareAsset, 10),
+                              ex_CHANGE_TRUST_TRUST_LINE_MISSING);
+
+            if (assetA.type() != ASSET_TYPE_NATIVE)
+            {
+                root.changeTrust(assetA, 10);
+
+                REQUIRE_THROWS_AS(
+                    root.changeTrust(poolShareAsset, 10),
+                    ex_CHANGE_TRUST_NOT_AUTH_MAINTAIN_LIABILITIES);
+                gateway.allowMaintainLiabilities(assetA, root);
+            }
+
+            if (assetB.type() != ASSET_TYPE_NATIVE)
+            {
+                // root is still missing assetB trustline
+                REQUIRE_THROWS_AS(root.changeTrust(poolShareAsset, 10),
+                                  ex_CHANGE_TRUST_TRUST_LINE_MISSING);
+                root.changeTrust(assetB, 10);
+
+                // assetB trustline is not authorized to maintain liabilities
+                REQUIRE_THROWS_AS(
+                    root.changeTrust(poolShareAsset, 10),
+                    ex_CHANGE_TRUST_NOT_AUTH_MAINTAIN_LIABILITIES);
+                gateway.allowMaintainLiabilities(assetB, root);
+            }
+
+            // this should create a LiquidityPoolEntry, and modify
+            // liquidityPoolUseCount on the asset trustlines
+            auto prePoolNumSubEntries = getNumSubEntries(root);
+            root.changeTrust(poolShareAsset, 10);
+
+            // TODO: This line requires the update to SponsorshipUtils
+            // REQUIRE(getNumSubEntries(root) - prePoolNumSubEntries == 2);
```

Comment:
This needs to be uncommented.

Reply:
Open an issue for this (no need to reference the issue from code), so that we don't forget to make this change

Reply:
(saw that Jon made the same comment in a different place)

Reply:
Opened the issue - https://github.com/stellar/stellar-core/issues/3156